### PR TITLE
Features/update announcements to match spec 178933309

### PR DESCRIPTION
**Pull request by rosalinekarr**
Problem
=======
We've updated the announcement spec since the SDK announcement module was written, so now the SDK is out of sync with the spec. Specifically, we're missing `createdAt` on all announcements.
[#178933309](https://www.pivotaltracker.com/story/show/178933309)

Solution
========
Added `createdAt` to all announcement types and updated tests accordingly

Change summary:
---------------
* Added `createdAt` to all DSNP announcement types
* Added an optional `createdAt` arg to DSNP factories
* Updated tests accordingly
* Also fixed GraphChange enum out of sync with spec

Steps to Verify:
----------------
1. Call a `create` method from the announcements module
1. Check that the resulting object has a `createdAt` field

### CODE REVIEW

**Review by amparks100 — APPROVED** [review]
looks good to me except for the duplicate createdAt's that github found 👍

**Comment by rosalinekarr** [comment]
> looks good to me except for the duplicate createdAt's that github found 👍

Fixed

File: src/core/announcements/factories.ts [hunk 1]
```diff
@@ -96,14 +107,17 @@ export interface ReactionAnnouncement extends TypedAnnouncement<AnnouncementType
  * @param   fromId    - The id of the user from whom the announcement is posted
  * @param   emoji     - The emoji to respond with
  * @param   inReplyTo - The DSNP Announcement Id of the parent announcement
+ * @param createdAt - Optional. The createdAt timestamp of the announcement as number of milliseconds since UNIX epoch
```
[the remark this answers is not in the document]
Comment (rosalinekarr) [reply src/core/announcements/factories.ts:111]: fixed

File: src/core/announcements/factories.ts [hunk 2]
```diff
@@ -113,8 +127,8 @@ export const createReaction = (
  * DSNPGraphChangeType: an enum representing different types of graph changes
  */
 export enum DSNPGraphChangeType {
+  Unfollow = 0,
```
[the remark this answers is not in the document]
Comment (rosalinekarr) [reply src/core/announcements/factories.ts:131]: This enum was out of sync with the spec. I've updated the changelog.

File: src/core/announcements/factories.ts [hunk 3]
```diff
@@ -40,11 +41,18 @@ export interface BroadcastAnnouncement extends TypedAnnouncement<AnnouncementTyp
...
-export const createBroadcast = (fromId: DSNPUserId, url: string, hash: HexString): BroadcastAnnouncement => ({
+export const createBroadcast = (
+  fromId: DSNPUserId,
+  url: string,
+  hash: HexString,
+  createdAt?: number
+): BroadcastAnnouncement => ({
   announcementType: AnnouncementType.Broadcast,
   contentHash: hash,
+  createdAt: createdAt || new Date().getTime(),
```
[the remark this answers is not in the document]
Comment (rosalinekarr) [reply src/core/announcements/factories.ts]: fixed

**Pull request merged**